libs/docs/src/docs/2.2.0/create-devfiles-with-templates.md

```diff
@@ -0,0 +1,186 @@
+            architectures: [ 'amd64' ]
+            displayName: Simple Web Service
+            icon: https://raw.githubusercontent.com/devfile-samples/devfile-stack-icons/main/golang.svg
+          ```
+    2. Setup `components`
+        - A `name` is required for a component
+        - If a `container` entity is defined, an `image` property must be 
+        specified
+        - Though `endpoints` is optional, it is needed to expose 
+        the port for web connections and requires: 
+          - An endpoint `name`
+          - A `targetPort` to expose, for example, `8080` if that is your http port
+        - The web service might need to connect to an external 
+        database using environment variables. In this case, define 
+        environment variable names and values for the 
+        container component under `env`:
+        ```yaml
+        components:
+          - name: web
+            container:
+              endpoints:
+                - name: http
+                  targetPort: 8080
+              env:
+                DATABASE_HOST: db.example.com
+                DATABASE_PORT: 5432
+                DATABASE_NAME: dev
+                DATABASE_USER: devuser
+                DATABASE_PASSWORD: devpassword
+              image: quay.io/devfile/golang:latest
+        ```
+    3. Adding `commands`
+        - An `id` to identify the command
+        - An `exec` entity must be defined with a `commandLine` string
+        and a reference to a `component` 
+          - *This implies that at least one component entity is defined 
+          under the root `components` element*
+        - The `workingDir` is set to where the project source is 
+        stored
+        - Command groups can be used to define automation that is useful
+        for executing a web service:
+        ```yaml
+        commands:
+          - id: build
+            exec:
+              commandLine: go build main.go
+              component: web
+              workingDir: ${PROJECT_SOURCE}
+              group:
+                kind: build
+                isDefault: true
+          - id: run
+            exec:
+              commandLine: ./main
+              component: web
+              workingDir: ${PROJECT_SOURCE}
+              group:
+                kind: run
+                isDefault: true
+        ```
+    4. Define starter projects
+        - Add a starter project under `starterProjects`
+        including at least a `name` and remote location, either
+        `git` or `zip`
+        - It is recommended to include a starter project 
+        description:
+          ```yaml
+          starterProjects:
+            - name: web-starter
+              description: A web service starting point.
+              git:
+                remotes:
+                  origin: https://github.com/devfile-samples/devfile-stack-go.git
+          ```
+    ```yaml {% title="Complete Web Service Template" filename="devfile.yaml" %}
```

Comment:
Should we have a title before this line describing the output? e.g. `Putting it all together, this is what the complete devfile looks like`